registration-processor/mosip-stage-executor/src/main/java/io/mosip/registration/processor/stages/executor/MosipStageExecutorApplication.java

```
@@ -0,0 +1,80 @@
+			StagesConfig stagesConfig = stageInfoApplicationContext.getBean(StagesConfig.class);
+			MutablePropertySources propertySources = stagesConfig.getCloudPropertySources();
+			
+			List<String> stageBeansBasePackages = StageClassesUtil.getStageBeansBasePackages(stagesConfig, propertySources);
+			
+			regProcLogger.info(">>>> Base packages for stage beans from configuration: {}", stageBeansBasePackages);
+			
+			List<Class<MosipVerticleAPIManager>> stageClasses = StageClassesUtil.getStageBeanClasses(stageBeansBasePackages);
+			
+			regProcLogger.info(">>>> Stage classes identified: {}", stageClasses.stream().map(Class::getCanonicalName).collect(Collectors.joining(", ")));
+
+			Class<?>[] entrypointConfigClasses = Stream.concat(Stream.of(StagesConfig.class), stageClasses.stream())
+					.toArray(size -> new Class<?>[size]);
+
+			try {
+				//This context should not be closed and to be kept for consumption by the verticles
+				AnnotationConfigApplicationContext mainApplicationContext = new PropertySourcesCustomizingApplicationContext(
+						entrypointConfigClasses) {
+							@Override
+							public MutablePropertySources getPropertySources() {
+								return propertySources;
+							};
+						};
+				
+				if (!stageClasses.isEmpty()) {
+					ExecutorService executorService = Executors.newFixedThreadPool(stageClasses.size());
+					stageClasses.forEach(stageClass -> executorService.execute(() -> {
+						try {
+							regProcLogger.info("Executing Stage: {}", stageClass.getCanonicalName());
+							MosipVerticleAPIManager stageBean = StageClassesUtil.getStageBean(mainApplicationContext, stageClass);
+							stageBean.deployVerticle();
+						} catch (Exception e) {
+							regProcLogger.error("Exception occured while loading verticles. Please make sure correct verticle name was passed from deployment script. \n {}",
+									ExceptionUtils.getStackTrace(e));
+						}
+					}));
+					executorService.shutdown();
+				}
```

In else case, lets throw an error with proper message and stop the application

---

Done.